pkg/registry/discovery/endpointslice/strategy.go

```diff
@@ -139,3 +145,37 @@ func dropDisabledFieldsOnUpdate(oldEPS, newEPS *discovery.EndpointSlice) {
 		}
 	}
 }
+
+// dropTopologyOnV1 on V1 request wipes the DeprecatedTopology field  and copies
+// the NodeName value into DeprecatedTopology
+func dropTopologyOnV1(ctx context.Context, oldEPS, newEPS *discovery.EndpointSlice) {
+	if info, ok := genericapirequest.RequestInfoFrom(ctx); ok {
+
+		// endpointListUpdated should be true for creates
+		endpointListUpdated := true
+		if oldEPS != nil {
+			endpointListUpdated = !apiequality.Semantic.DeepEqual(oldEPS.Endpoints, newEPS.Endpoints)
+		}
+
+		requestGV := schema.GroupVersion{Group: info.APIGroup, Version: info.APIVersion}
+		if requestGV == discoveryv1.SchemeGroupVersion {
```

Comment:
suggest checking our version and returning early for v1beta1 specifically. we'd want to drop deprecated topology the same way in future versions without needing to revisit this code.

```
func dropTopologyOnV1(ctx context.Context, oldEPS, newEPS *discovery.EndpointSlice) {
  if info, ok := genericapirequest.RequestInfoFrom(ctx); ok {
    // ... return early if v1beta1
  }
  ... check deepequal, drop, etc...
```

Reply:
that has a nice property of forcing us to look at this when v1b1 goes away

Reply:
updated so we check for v1beta1 requests and return early.